lib/internal/fs/utils.js

```diff
@@ -430,6 +430,17 @@ const validatePath = hideStackFrames((path, propName = 'path') => {
   }
 });
 
+let nonPortableTemplateWarn = true;
+
+function warnOnNonPortableTemplate(template) {
+  // Template strings passed to the mkdtemp() family of functions should not
+  // end with 'X' because they are handled inconsistently across platforms.
+  if (nonPortableTemplateWarn && template.endsWith('X')) {
+    process.emitWarning('mkdtemp() templates ending with X are not portable.');
```

Comment:
Would you mind expanding the message a bit that it's clear what this is about from the message itself? Without any further context it's difficult to follow what the warning is about.
We might add a reference to our documentation?

Reply:
Do you (or anyone else) have suggestions for specific wording? I agree that the message could be more detailed, but I'm struggling to come up with text that isn't too verbose for a runtime warning.

Reply:
Maybe use a recommendation that includes a link to the docs? 

```js
process.emitWarning(
  'It is recommended not to end `mkdtemp's` template prefix with an `X` \n' +
  'to guarantee portability. For details check:\n' +
  'https://nodejs.org/api/fs.html#fs_fs_mkdtemp_prefix_options_callback'
);
```

Reply:
Would you be OK with dropping the `#fs_fs_mkdtemp_prefix_options_callback` part? That seems prone to becoming out of date, and there are two other implementations (sync and promises).

Reply:
:+1: I'm good with that.